app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,34 @@
+class SessionsController < ApplicationController
+  def new
+    if current_user
+      flash[:notice] = "You are already logged in!"
+      login_user(User.find(session[:user]))
+    end
+  end
+
+  def create
+    user = User.find_by(email: params[:email])
+    if user && user.authenticate(params[:password])
+      session[:user] = user.id
+      flash[:notice] = "You are now logged in!"
+      login_user(user)
+    else
+      flash[:notice] = "You entered incorrect credentials."
+      render :new
+    end
+  end
+
+  def destroy
+    session.delete(:cart)
+    session.delete(:user)
+    flash[:notice] = "You have been sucessfully logged out. Bye!"
+    redirect_to "/"
+  end
+  private
+
+  def login_user(user)
```

Comment:
We might consider changing this method name to something more akin to what's occurring - login_redirect, user_redirect, etc.

Reply:
I think that would make more sense naming wise

Reply:
Let's touch that up in a future user story, I'll make a note of it.